frontend/packages/common-frontend/src/ConnectionState.ts

```diff
@@ -122,10 +122,24 @@ export class ConnectionStateStore {
   }
 }
 
+const isLocalhost = (hostname: string) => {
+  if (hostname === 'localhost') {
+    return true;
+  }
+  if (hostname === '[::1]') {
+    return true;
+  }
+  if (hostname.match(/^127\.0\.0\.\d+$/)) {
```

Comment:
For class 8 it should be something like ^127\.\d+\.\d+\.\d+$
I would say there's no need to strictly check for number between 0 and 255

Reply:
Right, it is 127.* and not 127.0.0.*. Fixed and added test